File: backend/Models/Property.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace backend.Models;

public class Property
{
    [Key]
    public Guid Id { get; set; }
    public string? Street { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? Country { get; set; }
    public string? PostalCode { get; set; }
    public decimal? Latitude { get; set; }
    public decimal? Longitude { get; set; }
    [NotMapped]
    public string Address => $"{Street}, {City}, {State} {PostalCode}";
    public bool IsBillingAddress { get; set; }
    public Guid CustomerId { get; set; }
    [JsonIgnore]
    public Customer? Customer { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/Property.ts
export type TProperty = {
  id: string;
  street?: string;
  city?: string;
  state?: string;
  country?: string;
  postalCode?: string;
  latitude?: number;
  longitude?: number;
  address: string;
  isBillingAddress: boolean;
  customerId: string;
  createdAt: string;
  updatedAt: string;
};

export type TAddProperty = {
  street?: string;
  city?: string;
  state?: string;
  country?: string;
  postalCode?: string;
  latitude?: number;
  longitude?: number;
  isBillingAddress?: boolean;
  customerId?: string;
};

export type TAddBilingProperty = {
  street?: string;
  city?: string;
  state?: string;
  country?: string;
  postalCode?: string;
  latitude?: number;
  longitude?: number;
  customerId?: string;
};

export type TUpdateProperty = {
  id: string;
  street?: string;
  city?: string;
  state?: string;
  country?: string;
  postalCode?: string;
  isBillingAddress?: boolean;
};
